storage/innobase/os/os0enc.cc

```diff
@@ -1182,21 +1181,20 @@ byte *Encryption::encrypt(const IORequest &type, byte *src, ulint src_len,
       ut_ad(m_klen == KEY_LEN);
       ut_ad(m_iv != nullptr);
 
-      /* Total length of the data to encrypt. */
-      ulint data_len;
-        if (m_type == KEYRING && page_type == FIL_PAGE_COMPRESSED) {
-            data_len = src_enc_len - DST_HEADER_SIZE;  // We need those 8 bytes for
-                                                // key_version and post-encryption
-                                                // checksum
-        } else if (m_type == KEYRING && !type.is_page_zip_compressed()) {
-            data_len = src_enc_len - DST_HEADER_SIZE - 4;  // For keyring encryption we do
-                                                    // not encrypt last four bytes
-                                                    // which are equal to the LSN
-                                                    // bytes in header
-            // So they are not encrypted anyways
-        } else {
-            data_len = src_enc_len - DST_HEADER_SIZE;
-        }
+    /* Total length of the data to encrypt. */
+    if (m_type == KEYRING && page_type == FIL_PAGE_COMPRESSED) {
+        data_len = src_enc_len - FIL_PAGE_DATA;  // We need those 8 bytes for
+                                            // key_version and post-encryption
+                                            // checksum
+    } else if (m_type == KEYRING && !type.is_page_zip_compressed()) {
+        data_len = src_enc_len - FIL_PAGE_DATA - 4;  // For keyring encryption we do
+                                                // not encrypt last four bytes
+                                                // which are equal to the LSN
+                                                // bytes in header
+        // So they are not encrypted anyways
+    } else {
+        data_len = src_enc_len - FIL_PAGE_DATA;
+    }
```

Comment:
Looks correct to me. We are now using src_enc_len (which is the compressed length)Two minor comments

1. Is there a keyring testcase that fails without this change?
2. Can you move the end comment to a regular comment (full line comment)

Reply:
1. Most of encryption suite tests, innodb.table_encrypt_2_keyring, encryption.keyring_and_mk_encryption_exclusions
2. Fixed